scripts/language_model/cache_language_model.py

```diff
@@ -0,0 +1,207 @@
+import math
+import os
+import sys
+import mxnet as mx
+import gluonnlp as nlp
+
+curr_path = os.path.dirname(os.path.abspath(os.path.expanduser(__file__)))
+sys.path.append(os.path.join(curr_path, '..', '..'))
+
+
+parser = argparse.ArgumentParser(description=
+                                 'MXNet Neural Cache Language Model on Wikitext-2.')
+parser.add_argument('--bptt', type=int, default=2000,
+                    help='sequence length')
+parser.add_argument('--model_name', type=str, default='awd_lstm_lm_1150',
+                    help='name of the pretrained language model')
+parser.add_argument('--gpus', type=str,
+                    help='list of gpus to run, e.g. 0 or 0,2,5. empty means using cpu.'
+                         '(using single gpu is suggested)')
+parser.add_argument('--window', type=int, default=2000,
+                    help='cache window length')
+parser.add_argument('--theta', type=float, default=0.662,
+                    help='mix between uniform distribution and '
+                         'cache softmax distribution over previous words')
+parser.add_argument('--lambdas', type=float, default=0.1279,
+                    help='linear mix between only cache (1) and only vocab (0) distribution')
+parser.add_argument('--path_to_params_file', type=str, default=None,
+                    help='path to the saved params file of user pre-trained model, '
+                         'including the params file, e.g., ~/.mxnet/models/awd_lstm_lm_1150.params')
+args = parser.parse_args()
+
+###############################################################################
+# Load vocabulary
+###############################################################################
+
+context = [mx.cpu()] if args.gpus is None or args.gpus == '' else \
+          [mx.gpu(int(x)) for x in args.gpus.split(',')]
+
+print(args)
+
+_, vocab = nlp.model.get_model(name=args.model_name,
+                               dataset_name='wikitext-2',
+                               pretrained=False,
+                               ctx=context)
+ntokens = len(vocab)
+
+###############################################################################
+# Build the cache model and load pre-trained language model
+###############################################################################
+
+
+if not args.path_to_params_file:
+    cache_cell = nlp.model.get_cache_model(name=args.model_name, dataset_name='wikitext-2',
+                                           window=args.window, theta=args.theta,
+                                           lambdas=args.lambdas, ctx=context)
+else:
+    model, _ = nlp.model.get_model(name=args.model_name,
+                                   dataset_name='wikitext-2',
+                                   pretrained=False,
+                                   ctx=context)
+    cache_cell = nlp.model.CacheCell(model, ntokens, args.window, args.theta, args.lambdas)
+    cache_cell.load_params(args.path_to_params_file, ctx=context)
```

Comment:
could you add a test for the interchangeable use of the parameters from a cached and non-cached model?